llvm/test/Transforms/InstCombine/saturating-add-sub.ll

```diff
@@ -2350,4 +2350,57 @@ define i8 @fold_add_umax_to_usub_multiuse(i8 %a) {
   ret i8 %sel
 }
 
+define i32 @uadd_with_zext(i32 %x, i32 %y) {
+; CHECK-LABEL: @uadd_with_zext(
+; CHECK-NEXT:    [[COND:%.*]] = call i32 @llvm.uadd.sat.i32(i32 [[X:%.*]], i32 [[Y:%.*]])
+; CHECK-NEXT:    ret i32 [[COND]]
+;
```

Comment:
Producing uadd.sat doesn't match the description

Reply:
Well, that's how instcombine is folding it.

What I do only generates the add and compare.

Other folds are folding it further that I don't have any part of

Reply:
The description of the test describes the input, I wasn't aware that it's meant to describe the output